src/distributions/mod.rs

```diff
@@ -82,6 +83,8 @@ pub struct Weighted<T> {
     pub item: T,
 }
 
+impl <T: Copy> Copy for Weighted<T> {}
```

Comment:
I think this should be `#[derive(Copy)]`-able too. (`#[derive]` will add a `Copy` bound to the type parameter in the impl.)